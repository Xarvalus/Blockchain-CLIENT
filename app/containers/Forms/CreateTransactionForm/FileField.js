import * as React from 'react'

import StyledInput from './StyledInput'

type Props = {
  id: string,
  input: {
    onChange: Function,
    name: string,
  },
}

export default class FileField extends React.PureComponent<Props> {
  onChange = (event: SyntheticInputEvent<HTMLInputElement>) => {
    const { input: { onChange } } = this.props

    onChange(event.target.files[0])
  }

  render(): React.Node {
    const { id, input: { name } } = this.props

    return (
      <StyledInput id={id} type="file" name={name} onChange={this.onChange} />
    )
  }
}
