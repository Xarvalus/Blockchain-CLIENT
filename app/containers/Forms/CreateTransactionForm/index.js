import * as React from 'react'
import { Field, reduxForm } from 'redux-form/immutable'

import FileField from './FileField'

type Props = { handleSubmit: Function }

@reduxForm({
  form: 'createAssetTransaction',
})
export default class CreateTransactionForm extends React.PureComponent<Props> {
  render(): React.Node {
    const { handleSubmit } = this.props

    return (
      <form onSubmit={handleSubmit}>
        { /* @TODO should be composed of reactstrap's Form components */ }

        <div className="form-group">
          <label htmlFor="name">Asset Name</label>
          <Field type="text" className="form-control" name="name" component="input" />

          <label htmlFor="author">Author</label>
          <Field type="text" className="form-control" name="author" component="input" />
        </div>

        <div className="form-group">
          <label htmlFor="type">Asset Category</label>
          <Field className="form-control" name="type" component="select">
            <option value="Painting">Painting</option>
            <option value="Drawing">Drawing</option>
            <option value="Sculpture">Sculpture</option>
          </Field>

          <label htmlFor="technique">Technique</label>
          <Field type="text" className="form-control" name="technique" component="input" />
        </div>

        <div className="form-group">
          <label htmlFor="price">Price estimation</label>
          <Field type="text" className="form-control" name="price" component="input" />

          <label htmlFor="auctionHouse">Auction House</label>
          <Field type="text" className="form-control" name="auctionHouse" component="input" />
        </div>
        <div className="form-group clearfix">
          <label htmlFor="preview">
            <span className="btn btn-primary mb-2 float-left">Upload preview</span>
          </label>
          <Field id="preview" className="form-control" name="preview" component={FileField} />

          <button type="submit" className="btn btn-primary mb-2 float-right">Submit</button>
        </div>
      </form>
    )
  }
}
