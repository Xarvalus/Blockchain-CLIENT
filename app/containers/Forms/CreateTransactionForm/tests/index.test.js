import React from 'react'
// import { shallow } from 'enzyme'
import ShallowRenderer from 'react-test-renderer/shallow'

import CreateTransactionForm from '../index'

describe('<CreateTransactionForm />', () => {
  // let wrapper
  // beforeEach(() => {
  //   wrapper = shallow(<CreateTransactionForm />)
  // })

  it('renders correctly', () => {
    const renderer = new ShallowRenderer()
    const snapshot = renderer
      .render(<CreateTransactionForm />)

    expect(snapshot).toMatchSnapshot()
  })
})
