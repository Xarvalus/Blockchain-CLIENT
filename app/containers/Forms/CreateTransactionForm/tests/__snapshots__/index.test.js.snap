// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CreateTransactionForm /> renders correctly 1`] = `
<Connect(Form(CreateTransactionForm))
  destroyOnUnmount={true}
  enableReinitialize={false}
  forceUnregisterOnUnmount={false}
  form="createAssetTransaction"
  getFormState={[Function]}
  initialValues={undefined}
  keepDirtyOnReinitialize={false}
  persistentSubmitErrors={false}
  pure={true}
  shouldAsyncValidate={[Function]}
  shouldError={[Function]}
  shouldValidate={[Function]}
  shouldWarn={[Function]}
  touchOnBlur={true}
  touchOnChange={false}
  updateUnregisteredFields={false}
/>
`;
